[FEATURE] Refonte du site vitrine: bannière sur la page d'accueil (PIX-1007)

celineung opened the pull request:
## :unicorn: Problème
La refonte du site vise à améliorer les points suivants:
- Le site n'est pas adapté aux écrans larges (>1280)
- Mettre à jour le contenu pour présenter Pix et tenir compte de l’évolution du projet (offre pro etc)
- Coherence de notre image vis à vis de la charte
- Rendre collaboratif le contenu pour le pôle com

## :robot: Solution
Cette PR concerne la refonte de la bannière de la page d'accueil.

## :rainbow: Remarques
- Le composant `banner` est réutilisable, il tire ses infos de prismic:
  - un `bannertitle`
  - un `bannercontent`
  - un `bannerImage`
  - un `bannerBackground`
- Le background-image peut ne pas apparaître si le resize se fait via le navigateur sans recharger la page
- Le test est KO car il dépends de la questions 1 ci-dessous

QUESTIONS :
- Pour la mise en production de Pix-site : est-ce qu'on fait morceaux par morceaux  OU tout le nouveau design de Pix Site d'un coup ?
- Voir avec Vincent, Gul pour la gestion des versions dans le premier cas ? 
- Devrait-on utiliser `css-grid-layout` (https://developer.mozilla.org/fr/docs/Web/CSS/CSS_Grid_Layout) ? -> Oui mais dans une prochaine PR "refacto" pour ne pas ralentir celle ci

- @QuentinChapelain-ui Devrait-on uniformiser les breakpoints entre pix-site et pix-app ?

Pour les tests :  npm run test -- -u


## :sparkles: Review App
https://pix-site-review-pr129.osc-fr1.scalingo.io

Anne-Gaelle-S committed:
Banner without style and images

Commit:
Handle large-screen devices

aceol commented on components/slices/banner.vue:
```diff
@@ -0,0 +1,44 @@
+<template>
+  <section>
+    <prismic-rich-text :field="title" />
+    <prismic-rich-text :field="textContent" />
```
Pour le formulaire de verification de code de certif, on aurait besoin d'un slot ici 
`<slot></slot>`

ce qui nous permettrais de faire un 
```
<banner :title... :...>
   <mon-formulaire-code-verif></mon-formulaire-code-verif>
</banner>
```

aceol commented on pages/index.vue:
```diff
@@ -1,5 +1,7 @@
 <template>
   <div class="index">
+    <banner :content="document[4]"></banner>
```
Ca serait chouette d'avoir des getter pour document[4] (et les autres), ca correspond à un champ particulier et ca serait beaucoup plus lisible

P-Jeremy commented on components/slices/banner.vue:
```diff
@@ -0,0 +1,254 @@
+<template>
+  <section :class="`${bannerWithBackgroundImageClass}`">
+    <div :class="`banner ${bannerWithImageClass}`">
```
On peut aussi faire comme ceci et s'affranchir de mettre la logique dans le js
```suggestion
    <div :class="['banner', {'banner--with-image': hasImage}]">
```

aceol commented on components/slices/banner.vue:
```diff
@@ -99,6 +102,18 @@ export default {
+        'banner-with-background'
+      )[0]
+      if (screen.width <= 1280) {
+        banner.style.background = `no-repeat url(${this.content.primary.bannerbackground.url})`
```
Pas possible de changer cette partie uniquement? (et laisser le reste en css)

celineung replied:
Comment ça?

aceol replied:
je suis pas certain de comprendre pourquoi on fait tout ca dans le javascript, à part la partie url de l'image qui est dynamique

Anne-Gaelle-S replied:
On n'a pas réussi à mettre l'url du background dynamiquement dans le css du coup on la met en js.
Je ne vois pas d'autres moyen de faire pour ça mais si tu as des suggestions on est prenneuses @aceol  :)

HEYGUL replied:
Je pense qu'@aceol veut dire qu'on pourrait tout faire en css, sauf la partie url qui est dynamique.

celineung replied:
on a finalement laissé le `backgroundSize` et `backgroundPosition` côté js, car le `background` redéfini ces deux paramètres à chaque fois

P-Jeremy commented on components/slices/banner.vue:
```diff
@@ -0,0 +1,254 @@
+    },
+  },
+  computed: {
+    bannerWithImageClass() {
+      return this.hasImage ? 'banner--with-image' : ''
+    },
```
https://github.com/1024pix/pix-site/pull/129#discussion_r463502664

Du coup, plus besoin de ce computed

```suggestion

```

Commit:
Extract prismic document into computed variables

Commit:
Add slot to banner

HEYGUL commented on components/slices/banner.vue:
```diff
@@ -0,0 +1,231 @@
+<template>
+  <section :class="`${bannerWithBackgroundImageClass}`">
```
Ne peut-on pas faire comme la `div` juste au-dessous ?
```suggestion
  <section :class="{ 'banner-with-background': hasBackgroundImage }">
```
Et du coup remplacer la propriété computed `bannerWithBackgroundImageClass` par `hasBackgroundImage`.

HEYGUL commented on components/slices/banner.vue:
```diff
@@ -0,0 +1,231 @@
+      return this.content.primary.bannerimage
+    },
+    fas() {
+      return fas
```
Est-ce nécessaire ? J'ai l'impression qu'on pourrait utiliser `fas` directement dans le template.

Anne-Gaelle-S replied:
Si on essaie d'utiliser "fas" directement dans le template en enlevant ce getter ça plante : 
<img width="589" alt="image" src="https://user-images.githubusercontent.com/38167520/89046315-0ea72900-d34d-11ea-9824-8e6672f74014.png">

<img width="589" alt="image" src="https://user-images.githubusercontent.com/38167520/89046269-0222d080-d34d-11ea-8227-88294b33d4d7.png">

HEYGUL commented on components/slices/banner.vue:
```diff
@@ -0,0 +1,231 @@
+    },
+  },
+  mounted() {
+    if (this.bannerWithBackgroundImageClass) {
```
D'un point de vue compréhension, il serait mieux d'avoir un test genre :
```suggestion
    if (this.hasBackgorundImage) {
```

HEYGUL commented on components/slices/banner.vue:
```diff
@@ -0,0 +1,231 @@
+      return link.bannerlinkurl.url.includes('pix-videos/')
+    },
+    videoClass(link) {
+      if (this.isVideo(link)) {
+        return 'banner__video'
+      }
+      return ''
```
```suggestion
      return this.isVideo(link) ? 'banner__video' : ''
```

celineung commented on assets/scss/globals/_colors.scss:
```diff
@@ -6,6 +6,8 @@ $grey-5: #666666;
 $grey-6: #444444;
 $grey-7: #DDDDDD;
 $grey-8: #96a0af;
+$grey-9: #6b778c;
```
pourquoi ne pas utiliser `darken` ou `lighten`

celineung replied:
il s'agit de couleur de boutons (non hover), du coup il serait préférable de les laisser

celineung commented on components/slices/banner.vue:
```diff
@@ -38,7 +42,51 @@ export default {
+
+  &__title h1 {
+    font-size: 46px;
+    line-height: 61px;
```
`rem`?

celineung commented on .vscode/launch.json:
```diff
@@ -0,0 +1,33 @@
+{
```
supprimer ce fichier?
et ajouter ce fichier au .gitignore (cf. celui de `pix-app`)

celineung commented on components/slices/banner.vue:
```diff
@@ -1,21 +1,23 @@
-        >
-          {{ link.bannerlinktext }}
-        </pix-link>
+  <section :class="[{ 'banner-with-background': hasBackgroundImage }]">
```
possibilité?
```suggestion
  <section :class="{ 'banner-with-background': hasBackgroundImage }">
```

celineung commented on components/slices/banner.vue:
```diff
@@ -43,12 +45,23 @@ export default {
-    },
+  },
+  mounted() {
+    if (this.hasBackgorundImage) {
```
```suggestion
    if (this.hasBackgroundImage) {
```

celineung commented on components/slices/banner.vue:
```diff
@@ -110,6 +132,7 @@ export default {
     height: 44px;
     margin: 0 10px;
     font-family: 'Roboto', Arial, sans-serif;
+    font-size: 16px;
```
`rem`?

celineung commented on components/slices/banner.vue:
```diff
@@ -74,13 +74,18 @@ export default {
       banner.style.background = `center no-repeat url(${this.content.primary.bannerbackground.url})`
     }
   },
+  beforeDestroy() {
```
supprimer le `beforeDestroy`

Anne-Gaelle-S replied:
Finalement on s'est rendue compte qu'on avait besoin de l'eventlister pour qu'au redimensionnement de la page le fond change.

celineung commented on components/slices/banner.vue:
```diff
@@ -91,6 +96,18 @@ export default {
+      const banner = document.getElementsByClassName(
+        'banner-with-background'
+      )[0]
+      if (screen.width <= 1280) {
```
sortir `1280` dans une constante

celineung replied:
voir s'il existe un service `media` comme dans `ember-responsive`

celineung replied:
On a pas trouvé de service équivalent côté nuxt/vue.
On a pour l'instant sorti la variable `1280` dans un fichier js de constantes.

celineung commented:
Remarques de Github actions:
> The "asyncData" property should be above the "computed" property on line 52

Commit:
BSR: remove fontawesome import

Commit:
Add style to banner

Commit:
Add image on condition

An image can be added on the right side of the text if the image is provided in
prismic

HEYGUL commented on components/slices/banner.vue:
```diff
@@ -37,7 +37,6 @@
 
 <script>
 /* eslint-disable-next-line no-unused-vars */
```
a-t-on encore besoin de désactiver le warning eslint ?

Anne-Gaelle-S replied:
non :)

Anne-Gaelle-S committed:
Add .vscode files in gitignore

Anne-Gaelle-S committed:
Add background-image on condition

Background-image is displayed if it's provided in prismic.

Anne-Gaelle-S committed:
Add video pop up

Anne-Gaelle-S committed:
BSR: asyncData order

Commit:
Fix banner style
